### PR TITLE
Pandas dataframe backend

### PR DESCRIPTION
**Pull request by Jacquelinet19**
# Reliquery Pull Request
**Thank you for contributing to Reliquery.** 
Before submitting this PR please make sure:

### PR checklist
- [x] Your code builds clean without any errors or warnings
- [x] Code is formatted using the supported linter locally before pushing changes
- [x] You have added unit tests where applicable 
- [x] All tests complete and pass with changes locally
- [x] Have updated the documentation accordingly.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe proposed changes
- Updated readme to include pandas info 
- Take in pandas and use json for serialization

### CODE REVIEW

File: reliquery/relic.py [hunk 1]
```diff
@@ -252,13 +253,43 @@ def add_json(self, name: str, json_data: Dict) -> None:
...
+        self.storage.put_text(
+            [self.relic_type, self.name, "pandasdf", name], json_pandasdf
+        )
+        self._add_metadata(metadata)
+
+    def list_pandasdf(self):
+        return self.storage.list_keys([self.relic_type, self.name, "pandasdf"])
+
+    def get_pandasdf(self, name: str):
+        self.assert_valid_id(name)
```
Comment (AntiAnt) [inline reliquery/relic.py:285]: We should probably include return types in our method definitions.

**Review by AntiAnt — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -78,6 +78,24 @@ Get json by taking the name and returning the dictionary
 r.get_json("json")
 ```
 
+### Pandas DataFrame supported
+Note that json is used to serialize which comes with other caveats that can be found here: https://pandas.pydata.org/pandas-docs/version/0.23/generated/pandas.DataFrame.to_json.html
```
Comment (ibebrett) [inline README.md:82]: I think this makes sense to include in the docstring for the function itself too.

Info about doc strings:

https://www.python.org/dev/peps/pep-0257/

We can follow up on another PR though.

File: reliquery/relic.py [hunk 3]
```diff
@@ -252,13 +253,43 @@ def add_json(self, name: str, json_data: Dict) -> None:
     def list_json(self) -> List[str]:
         return self.storage.list_keys([self.relic_type, self.name, "json"])
 
-    def get_json(self, name: str) -> dict:
+    def get_json(self, name: str) -> Dict:
```
Comment (ibebrett) [inline reliquery/relic.py:256]: One thing about typing. We can fix this later, but in general json can be a list, a dict, or a string or number. We can figure out the typing for this later as well.

**Review by ibebrett — APPROVED** [review]
Great work all!

**Pull request merged**